lib/fog/libvirt/compute.rb

```diff
@@ -57,12 +57,16 @@ def initialize(options={})
           require 'libvirt'
         end
 
-        private
-
         def client
-          return @client if defined?(@client)
+          @client ||= Class.new do
+            def libversion
+              1002009
```

Comment:
Yes. you would be wrong. =)

The Mock class is not for test purposes directly but rather a feature fog has so you can try your code without actually accessing the provider api (It stores all data inside objects so you can get the result on a later call).

Your proposed solution (Set it from within the test code itself) would fit better. =)